src/lightkurve/lightcurve.py

```diff
@@ -1382,7 +1382,9 @@ def bin(
             ts.remove_column(colname)
             ts.add_column(tmpcol, name=colname, index=idx)
 
-        return self.__class__(ts)
+        binned_lc = self.__class__(ts)
+        binned_lc.meta.update(self.meta)
+        return binned_lc
```

Comment:
```suggestion
        return self.__class__(ts, meta=self.meta)
```
Might do it even as concisely as the original, though that is perhaps a matter of taste. But when adding it afterwards I see a
```
UserWarning: Lightkurve doesn't allow columns or meta values to be created via a new attribute name.A new attribute is created. It will not be carried over when the object is copied. - see https://docs.lightkurve.org/api/lightkurve.lightcurve.LightCurve.html
```

Reply:
I prefer the shortened version you suggested as well.

Regarding the `UserWarning`, I did not see it when I ran the test. 

I did notice an existing unrelated `test_attr_access_others()` does emit the warning. Could it be possible the one you saw came from `test_attr_access_others()`?

Reply:
I did not run the tests for this branch, just got the warning when adding a `meta` in normal code. Could be that they are marked ignore in the tests.

Reply:
I am not sure what you mean by "when adding a `meta` in normal code"...

The change you suggested, `self.__class__(ts, meta=self.meta)`, does not cause warnings.  

I guess you noticed the (spurious) warning emitted by another test `test_attr_access_others()`. The issue has just been fixed in PR #1045.

Reply:
No, again, I did not run any tests whatsoever, just made a simple check to verify the meta is lost in binning
```python
>>> lc = lk.LightCurve(time=[1, 2, 3, 4], flux=[1, 1, 1, 1])
>>> lc.meta = {'TELESCOP': 'Kepler'}
<stdin>:1: UserWarning: Lightkurve doesn't allow columns or meta values to be created via a new attribute name.A new attribute is created. It will not be carried over when the object is copied. - see https://docs.lightkurve.org/api/lightkurve.lightcurve.LightCurve.html
>>> lb = lc.bin(2)
>>> lb.meta
OrderedDict()
```
Note also that it explicitly warns that it will not be preserved on copying, so that might have been designed deliberately at some point; however the docs link is broken, and I did not see any other information on that so far.
But I don't see a strong reason not to copy it over to the binned lightcurve (apart perhaps from some information that would only apply to the original, like number of points...).
Regarding other stats, `time_bin_start` and `time_bin_size` are already stored; others like no. of points per bin may not be so easy to get out of `aggregate_downsample`.

Reply:
1. On the UserWarning upon `lc.meta = {'TELESCOP`, 'Kepler'}, it is a bug that is just filed as #1046 . The warning is for intended for users creating their own  custom attribute, e.g., `lc.some_new_attr = 123`, and should not appear when dealing with built-in `lc.meta`

2. Reasons for copying metadata over binned lightcurve. Some LightCurve functions require metadata, e.g., `query_solar_system_object()`, `plot()`, etc.

3. Regarding other stats, I didn't realize `time_bin_size` is already there. I mistakenly looked for it in `lc.meta['time_bin_size']`. So I guess we're okay.